src/request.ts

```diff
@@ -0,0 +1,54 @@
+import { AxiosError } from "axios";
+
+export class ExecuteRequestError extends Error {
+  originalError: AxiosError;
+  responseStatus: number | null;
+  responseMessage: string | null;
+
+  constructor(message: string, axiosError: AxiosError, responseStatus: number | null, responseMessage: string | null) {
+    super(message);
+    this.originalError = axiosError;
+    this.responseStatus = responseStatus;
+    this.responseMessage = responseMessage;
+
+    // Required for `instanceof` to work.
+    Object.setPrototypeOf(this, ExecuteRequestError.prototype);
+  }
+}
+
+/**
+ * Gets the full, descriptive error response returned from skyd on the portal.
+ *
+ * @param err - The Axios error.
+ * @returns - A new error if the error response is malformed, or the skyd error message otherwise.
+ */
+export function getPortalErrResponse(err: AxiosError): ExecuteRequestError {
```

Comment:
Wouldn't it have been cleaner to make this the `constructor`? 
If that is weird for whatever reason then perhaps a `static` so we could do `ExecuteRequestError.From(axiosErr)`?